streams-tests/shared/src/test/scala/zio/stream/experimental/ZSinkSpec.scala

```diff
@@ -530,6 +530,92 @@ object ZSinkSpec extends ZIOBaseSpec {
           )
         }
       ),
+      suite("fromQueue")(
+        test("should enqueue all elements") {
+
+          for {
+            queue          <- ZQueue.unbounded[Int]
+            _              <- ZStream(1, 2, 3).run(ZSink.fromQueue(queue))
+            enqueuedValues <- queue.takeAll
+          } yield {
+            assert(enqueuedValues)(hasSameElementsDistinct(List(1, 2, 3)))
+          }
+
+        }
+      ),
+      suite("fromQueueWithShutdown")(
+        test("should enqueue all elements and shutsdown queue") {
+
+          def createQueueSpy[A](q: Queue[A]) = new ZQueue[Any, Any, Nothing, Nothing, A, A] {
```

Comment:
Is this necessary? Can't you just create a queue and check if it was shut down with it's `isShutdown` method?

Reply:
I generally like to test the whole behaviour from users perspective and thing they care about it that all elements were enqueued and then the queue was shut down. Sadly was not able to find a more elegant/reliable way to get all the enqueued elements, after shutdown occurs one cannot dequeue anymore. Would have wanted to write a similar test for Hub as well. If you think it is a overkill in this case I can scrap it and convert it to similar to  `fromHubWithShutdown`, but personally I think more testing the better :)

Reply:
Ok I understand now. Let's keep it :)